### DISCUSSION TITLE
Tag convention: `v1.17.0` or `v1.17.0.0`?

### DISCUSSION THREAD

**Discussion by nbolton** [deskflow/deskflow#7843] [Development]
Quick thought: We're currently using `v1.17.0` for our release name and tag name, dropping the 4th part or "tweak" number, as @sithlord48 calls it, to indicate that it is not a continuous build. For the benefit of packagers, the "tweak" is synonymous with "build" or "revision".

![image](https://github.com/user-attachments/assets/0ad8d9a8-ca66-4cc2-9eec-187200521cda)

Does this lack of 4th digit on the release page make the version ambiguous? Should we perhaps always include the 4th digit to clarify that it is the stable tag? If so, should both the tag and release always include the 4th digit? `v1.17.0.0`

I'm happy either way, but I believe consistency and documented steps here are important for packagers to follow what's going on. Many thanks to our packagers for bearing with us while we figure out our procedures for our new project! :slightly_smiling_face:

**Comment by cdwmhcc** [comment]
It should be `1.17.0.0` instead of 1.17.0
Part 4 cannot be ignored!

Additionally, the tag name should contain the version number. ex: `continuous-1.17.0.167`

![image](https://github.com/user-attachments/assets/83fcf953-d71e-41a6-82a3-092070ce040f)

One more very important thing, please do not replace old files from the released version.

**Comment by sithlord48** [comment]
You tag V 1.17.0 NEVER tag with the fourth number that is expected to be generated its the number of commits since the last tag

**Reply by sithlord48** [reply]
Also the tag needs to match the "Real" tag

**Comment by sithlord48** [comment]
> You tag V 1.17.0 NEVER tag with the fourth number that is expected to be generated its the number of commits since the last tag


I feel Like this might need a bit of explanation about the version generation.

when you open the deskflow project and cmake starts to parse the project the first thing is does is look for git and if found we call `git describe` to describe for us the state of the project. That is gonna return something like this  `v1.17.0-169-g16a1ba8f`

Now that we have this we use some regular expressions to match (cmake supports a limited set of regex tokens )
 - our Major `1`  , Regex is `^[0-9]+`  (we remove the v the before processing)
 - our Minor `17` Regex is `^[0-9]+\.[0-9]+`  (then remove the first bit) 
 - our patch `0`    Regex is `\.[0-9]+\-[0-9]+` (then remove the first bit) 
 - and our tweak `169`. Regex is `\-[0-9]+\-`  (removing the two - after) 
    -   The tweak is a special value and it is the number of commits since our last tag of `v*.*.*` 
    -   some schema will prefix this with `r`

If we were to tag `v1.17.1.0` our describe would return  `v1.17.1.0-0-gSHA`
With this input we generate the version:  `Building deskflow-1.17.0.0`
But we know that is incorrect since we would have expected: `Building deskflow-1.17.1.0`

**Reply by marlop352** [reply]
if any tagged fix/patch is released as "x.y.z.0" then I think there is no harm in not including the ending ".0" in my opinion
